scipy/sparse/_construct.py

```diff
@@ -315,15 +399,21 @@ def eye(m, n=None, k=0, dtype=float, format=None):
     format : str, optional
         Sparse format of the result, e.g., format="csr", etc.
 
+    .. warning::
+
+        This function returns a sparse matrix -- not a sparse array.
+        You are encouraged to use ``eye_array`` to take advantage
+        of the sparse array functionality.
+
     Examples
     --------
     >>> import numpy as np
-    >>> from scipy import sparse
-    >>> sparse.eye(3).toarray()
+    >>> import scipy as sp
+    >>> sp.sparse.eye(3).toarray()
     array([[ 1.,  0.,  0.],
            [ 0.,  1.,  0.],
            [ 0.,  0.,  1.]])
-    >>> sparse.eye(3, dtype=np.int8)
+    >>> sp.sparse.eye(3, dtype=np.int8)
     <3x3 sparse matrix of type '<class 'numpy.int8'>'
         with 3 stored elements (1 diagonals) in DIAgonal format>
 
```

Comment:
I'd prefer to avoid duplication of implementation for eye + eye_array. Maybe we can do something similar to `_random()` below?